internal/vendors/service.go

```diff
@@ -190,6 +194,62 @@ func (v *VendorService) replacePlaceholder(template string, replacements map[str
 	return template
 }
 
+func (v *VendorService) GetPopulatedEmailStatus(
```

Comment:
Honestly, I still don't get where does this one used and the GetAllEmailStatus used in FE.